docs/sphinx/using/examples/measuring_kernels.rst

```diff
@@ -0,0 +1,85 @@
+Measuring Kernels
+======================
+
+.. tab:: Python
+
+    .. literalinclude:: ../../examples/python/measuring_kernels.py
+        :language: python
+        :start-after: [Begin Docs]
+        :end-before: [End Docs]
+
+.. tab:: C++
+
+    .. literalinclude:: ../../examples/cpp/measuring_kernels.cpp
+        :language: cpp
+        :start-after: [Begin Docs]
+        :end-before: [End Docs]
+
+Kernel measurement can be specified in the Z, X, or Y basis using `mz`, `mx`, and `my`. If a measurement is specified with no argument, the entire kernel is measured in that basis. Measurement occurs in the Z basis by default.
```

Comment:
This sentence is unclear: "If a measurement is specified with no argument". 
Is `measurement` here referring to the `mz`/`mx`/`my` instructions (we cannot call them w/o arguments) or the runtime `cudaq::sample` function? 
If the latter is the case, we might want to write like "If the kernel contains no measurement instructions, ..." instead.